src/barChart.ts

```diff
@@ -529,85 +570,196 @@ export class BarChart implements IVisual {
-                        },
-                        altConstantValueSelector: barDataPoint.selectionId.getSelector(),
-                        selector: dataViewWildcard.createDataViewWildcardSelector(dataViewWildcard.DataViewWildcardMatchingOption.InstancesAndTotals)
-                    });
-                }
-                break;
-            case 'generalView':
-                objectEnumeration.push({
-                    objectName: objectName,
-                    properties: {
-                        opacity: this.barChartSettings.generalView.opacity,
-                        showHelpLink: this.barChartSettings.generalView.showHelpLink
-                    },
-                    validValues: {
-                        opacity: {
-                            numberRange: {
-                                min: 10,
-                                max: 100
-                            }
-                        }
-                    },
-                    selector: null
-                });
-                break;
-            case 'averageLine':
-                objectEnumeration.push({
-                    objectName: objectName,
-                    properties: {
-                        show: this.barChartSettings.averageLine.show,
-                        displayName: this.barChartSettings.averageLine.displayName,
-                        fill: this.barChartSettings.averageLine.fill,
-                        showDataLabel: this.barChartSettings.averageLine.showDataLabel
-                    },
-                    selector: null
-                });
-                break;
+    private getColorSelectorShortcuts(subSelections: CustomVisualSubSelection[]): VisualSubSelectionShortcuts {
+        const selector = subSelections[0].customVisualObjects[0].selectionId?.getSelector();
+        return [
+            {
+                type: VisualShortcutType.Reset,
+                relatedResetSourceUids: [{
+                    sourceUid: colorSelectorReferences.fillSliceUid,
+                    selector
+                }]
+            },
+            {
+                type: VisualShortcutType.Navigate,
+                destinationUids: [{ cardUid: colorSelectorReferences.cardUid }],
+                label: 'colorlabel'
+            }
+        ];
+    }
+
+    private getColorSelectorStyles(subSelections: CustomVisualSubSelection[]): SubSelectionStyles {
+        const cardUid = colorSelectorReferences.cardUid;
+        const groupUid = colorSelectorReferences.groupUid;
+        const sliceUid = colorSelectorReferences.fillSliceUid;
+        const selector = subSelections[0].customVisualObjects[0].selectionId?.getSelector();
+        return {
+            type: SubSelectionStylesType.Shape,
+            fill: {
+                label: 'Visual_Fill',
+                reference: {
+                    cardUid,
+                    groupUid,
+                    sliceUid,
+                    selector,
+                },
```

Comment:
Let's replace it with an object
``` typescript
const reference = {
    cardUid: colorSelectorReferences.cardUid,
    groupUid: colorSelectorReferences.groupUid,
    sliceUid: colorSelectorReferences.sliceUid,
    selector: subSelections[0].customVisualObjects[0].selectionId?.getSelector(),
},
```
or simply: (depends on purpose of colorSelectorReferences)
``` typescript
const reference = {
    ...colorSelectorReferences,
    selector: subSelections[0].customVisualObjects[0].selectionId?.getSelector(),
},
```
and then return it
``` typescript
 fill: {
                label: 'Visual_Fill',
                reference
}
```